File: src/main/java/com/hypherionmc/orion/utils/Environment.java
/*
 * This file is part of orion, licensed under the MIT License (MIT).
 *
 * Copyright (c) 2024 HypherionSA and Contributors
 *
 */
package com.hypherionmc.orion.utils;

import org.jetbrains.annotations.Nullable;

import java.util.HashMap;

/**
 * @author HypherionSA
 * Custom Environment Variables
 */
public class Environment {

    private static final HashMap<String, String> variables = new HashMap<>();

    static void setVariables(HashMap<String, String> variables) {
        Environment.variables.clear();
        Environment.variables.putAll(variables);
    }

    @Nullable
    public static String getenv(String key) {
        return variables.getOrDefault(key, System.getenv(key));
    }

}

File: src/main/kotlin/com/hypherionmc/orion/utils/Environment.kt
/*
 * This file is part of orion, licensed under the MIT License (MIT).
 *
 * Copyright (c) 2025 HypherionSA and Contributors
 *
 */
package com.hypherionmc.orion.utils

/**
 * @author HypherionSA
 *
 * Expose a virtual environment variables environment, that contain both System and Doppler variables
 */
object Environment {

    // Internal map of Doppler and System environment variables
    private val variables: HashMap<String, String> = hashMapOf()

    /**
     * Internal method to add Doppler tokens to Environment Variables
     *
     * @param variables The Key-Value set of variables to install
     */
    internal fun setVariables(variables: HashMap<String, String>) {
        this.variables.clear()
        this.variables.putAll(variables)
    }

    /**
     * Get an environment variable from either Doppler or the System
     *
     * @param key The key to get the value of
     */
    @JvmStatic
    fun getenv(key: String): String? {
        return variables.getOrDefault(key, System.getenv(key))
    }
}